index.js

```diff
@@ -116,9 +109,9 @@ function mergeItems(itemType, currentItem, newItem, ignoreLocations) {
     return currentItem;
 }
 
-function subscribeOnParserEvents(parser, options, version, resolve, reject) {
+function subscribeOnParserEvents(parser, ignoredVisibilities, detectedVersion, resolve, reject) {

```

Comment:
Not sure that this changes are require there. Can you please explain your motivation for that case?

Reply:
I think you are referring to line 13-23 and 112-162? My motivation was simply refactoring as I tend to refactor as I read code. But I'm also new to open source, so I'm sorry if I was overzealous with the refactoring. I can revert the changes no problem.

Reply:
I'm just worry for this method signature.
Contextually at this point is not matter how component version will be taken, so the `version` or `componentVersion` can be used w/o any issues or misundestanding. `detectedVersion` mean that this version was computed automatically by `detectVersion` method, but actually that version can be directly pass by the API user.

Reply:
You are right, the name is wrong and confusing. I'll revert it back to `version`.

Reply:
And it's good thing we discussed this, because I made a mistake on line 198: I was not using `options.structure`.

That made me realize that there is no integration tests for the parse method auto-detect feature. `SvelteVersionDetector.detectVersionFromStructure` is tested directly in `integration/detector/detector.spec.js` but never as part of the typical workflow (calling `index.js@parse`). I added 2 simple tests to do that. Check [this commit](https://github.com/alexprey/sveltedoc-parser/pull/44/commits/a438a40627bb81b83474465fdf0b63b856e53958) for the new changes.

Reply:
Great, thanks!